cdap-proto/src/main/java/co/cask/cdap/proto/id/EntityId.java

```diff
@@ -163,4 +164,15 @@ protected static String remaining(Iterator<String> iterator, String fieldName) {
   private static String remaining(Iterator<String> iterator) {
     return Joiner.on(IDSTRING_PART_SEPARATOR).join(iterator);
   }
+
+  public Iterable<EntityId> getHierarchy() {
+    Vector<EntityId> hierarchy = new Vector<>();
```

Comment:
Does this hierarchy need to be generated every time the call is made? Curious, if it can be precomputed?